### PR TITLE
[MINI-4195] Add Custom Permission Attachment

### PR DESCRIPTION
**Pull request by laubengaier**
# Description
Add Custom Permission `Attachment` + in case of a base64 string check if the permission is granted or not.

## Links
https://jira.rakuten-it.com/jira/browse/MINI-4195

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: MiniApp/Classes/ui/MiniAppViewController.swift [hunk 1]
```diff
@@ -221,11 +221,7 @@ extension MiniAppViewController: MiniAppNavigationDelegate {
 
     /// will be used soon when knowing the cases to react to
     public func miniAppNavigation(shouldOpen url: URL, with responseHandler: @escaping MiniAppNavigationResponseHandler) {
-//        MiniAppExternalWebViewController.presentModally(url: url, externalLinkResponseHandler: { url in
-//            if url.absoluteString == "miniapp://close" {
-//                // dismiss
-//            }
-//        })
+        // TODO: Implement Navigation Handling when necessary
```
Comment (Climbatize) [inline MiniApp/Classes/ui/MiniAppViewController.swift:224]: nooooooo! 🤣

Comment (laubengaier) [reply MiniApp/Classes/ui/MiniAppViewController.swift:224]: dunno what happend here 😂 🤔

File: MiniApp/Classes/core/Display/RealMiniAppView.swift [hunk 2]
```diff
@@ -192,6 +192,19 @@ internal class RealMiniAppView: UIView {
...
+                    if
+                        let miniAppId = miniAppId,
+                        let _ = MiniApp.shared()
+                            .getCustomPermissions(forMiniApp: miniAppId)
+                            .filter({ $0.permissionName == .attachments && $0.isPermissionGranted == .allowed })
+                            .first
+                    {
+                        self.navigationDelegate?.miniAppNavigation(shouldOpen: requestURL, with: { (url) in
+                            self.webView.load(URLRequest(url: url))
+                        })
```
Comment (rleojoseph) [inline MiniApp/Classes/core/Display/RealMiniAppView.swift:205]: Guess we need to do bit of formatting here.

File: MiniApp/Classes/core/JavascriptBridge/MiniAppJavaScriptEnumerations.swift [hunk 3]
```diff
@@ -52,6 +52,8 @@ public enum MiniAppCustomPermissionType: String, Codable, CaseIterable {
     case deviceLocation = "rakuten.miniapp.device.LOCATION"
     /// Custom permission for retrieving points from the host app
     case points = "rakuten.miniapp.user.POINTS"
+    /// Custom permission for downloading attachmens
+    case attachments = "rakuten.miniapp.user.ATTACHMENTS"
```
Comment (rleojoseph) [inline MiniApp/Classes/core/JavascriptBridge/MiniAppJavaScriptEnumerations.swift:56]: Guess we had to wait for finalizing the permission name

File: MiniApp/Classes/core/Display/RealMiniAppView.swift (hunk 2 above)
Comment (Climbatize) [reply MiniApp/Classes/core/Display/RealMiniAppView.swift]: also, instead of `let _ =` please use `!= nil`

**Comment by rleojoseph** [comment]
> > Looks good to me 👍 Just Switlint fix is needed.
> 
> @rleojoseph yeah already wanted to ask about this, there is a 200 char switlint error which doesn't make sense for the base64 string (I've already used a 1x1 black pixel) which results in >200. I can split the string into to parts to avoid the lint part but dunno

@laubengaier  Multi-line or `swiftlint:disable:next line_length` didn't work ??

**Pull request merged**